app/src/main/java/de/danoeh/antennapod/ui/screen/feed/preferences/FeedSettingsFragment.java

```diff
@@ -462,42 +462,54 @@ private void setupKeepUpdatedPreference() {
             });
         }
 
-        private void setupAutoDownloadGlobalPreference() {
-            if (!UserPreferences.isEnableAutodownload()) {
-                SwitchPreferenceCompat autodl = findPreference("autoDownload");
-                autodl.setChecked(false);
-                autodl.setEnabled(false);
-                autodl.setSummary(R.string.auto_download_disabled_globally);
-                findPreference(PREF_EPISODE_FILTER).setEnabled(false);
-            }
-        }
-
         private void setupAutoDownloadPreference() {
-            SwitchPreferenceCompat pref = findPreference("autoDownload");
-
-            pref.setEnabled(UserPreferences.isEnableAutodownload());
-            if (UserPreferences.isEnableAutodownload()) {
-                pref.setChecked(feedPreferences.getAutoDownload());
-            } else {
-                pref.setChecked(false);
-                pref.setSummary(R.string.auto_download_disabled_globally);
-            }
-
-            pref.setOnPreferenceChangeListener((preference, newValue) -> {
-                boolean checked = Boolean.TRUE.equals(newValue);
-
-                feedPreferences.setAutoDownload(checked);
+            findPreference(PREF_AUTODOWNLOAD).setOnPreferenceChangeListener((preference, newValue) -> {
+                switch ((String) newValue) {
+                    case "global":
+                        feedPreferences.setAutoDownload(FeedPreferences.AutoDownloadSetting.GLOBAL);
+                        break;
+                    case "always":
```

Comment:
always/never is a bit unexpected here. I would just call it enabled/disabled, like the setting is already called in Java